### PR TITLE
rgw_sal_motr:[CORTX-34158] Fix PutObject tagging, GetObject tagging and DeleteObject tagging versioning discrepancies.

### PR DESCRIPTION
**Pull request by shriya-deshmukh**
Problem:
Http_status code and error message discrepancies with AWS.
1. put-object-tagging, get-object-tagging and delete-object-tagging
    with invalid version-id. (AWS=400, MGW=404)
3. put-object-tagging, get-object-tagging and delete-object-tagging
    on delete-marker.(AWS=405, MGW=404)

Solution:
1. Return the 400 status code and error message the same as AWS for put-object-tagging, 
    get-object-tagging, and delete-object-tagging with invalid version-id.
2. Return the 405 status code and error message the same as AWS for put-object-tagging,
    get-object-tagging, and delete-object-tagging on delete marker.
 
Signed-off-by: Shriya Deshmukh <shriya.deshmukh@seagate.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Review by priyanka25081999 — APPROVED** [review]

File: src/rgw/rgw_sal_motr.cc [hunk 1]
```diff
@@ -1681,17 +1681,23 @@ MotrObject::~MotrObject() {
 //    return read_op.prepare(dpp);
 //  }
 
-int MotrObject::fetch_obj_entry_and_key(const DoutPrefixProvider* dpp, rgw_bucket_dir_entry& ent, std::string& bname, std::string& key, rgw_obj* target_obj)
+int MotrObject::fetch_obj_entry_and_key(const DoutPrefixProvider* dpp, rgw_bucket_dir_entry& ent, std::string& bname, std::string& key, rgw_obj* target_obj, RGWObjectCtx* rctx)
 {
-  int rc = this->get_bucket_dir_ent(dpp, ent);
+  req_state* s = static_cast<req_state*>(rctx->get_private());
```
Comment (cdeshmukh) [inline src/rgw/rgw_sal_motr.cc:1686]: Check [this](https://github.com/ceph/ceph/pull/44616#issuecomment-1248921018) comment, we can additionally check if req_state is indeed present and then only use it

File: src/rgw/rgw_sal_motr.cc [hunk 2]
```diff
@@ -1681,17 +1681,23 @@ MotrObject::~MotrObject() {
 //    return read_op.prepare(dpp);
 //  }
 
-int MotrObject::fetch_obj_entry_and_key(const DoutPrefixProvider* dpp, rgw_bucket_dir_entry& ent, std::string& bname, std::string& key, rgw_obj* target_obj)
+int MotrObject::fetch_obj_entry_and_key(const DoutPrefixProvider* dpp, rgw_bucket_dir_entry& ent, std::string& bname, std::string& key, rgw_obj* target_obj, RGWObjectCtx* rctx)
```
Comment (cdeshmukh) [inline src/rgw/rgw_sal_motr.cc:1684]: Can you put function arguments in multiple lines line in [MotrObject::copy_object_same_zone()](https://github.com/Seagate/cortx-rgw/blob/main/src/rgw/rgw_sal_motr.cc#L2383) , its growing and doesnt look proper

**Comment by shailesh-vaidya** [comment]
run vm-deployment

**Comment by shailesh-vaidya** [comment]
retest this please

**Pull request closed without merging**